Source/Modules/php.cxx

```diff
@@ -3038,17 +3118,25 @@ class PHP : public Language {
     String *iname = GetChar(n, "sym:name");
     ParmList *l = Getattr(n, "parms");
 
+    Delitem(class_need_free, Len(class_need_free) - 1);
+    Append(class_need_free, "1");
+
+    String *name_prefix = NewString("delete_");
+    String *intermediate_name = NewString(iname);
+    Replace(intermediate_name, name_prefix, "", DOH_REPLACE_FIRST);
+
     String *destructorname = NewStringEmpty();
     Printf(destructorname, "_%s", Swig_name_wrapper(iname));
     Setattr(classnode, "destructor", destructorname);
 
     Wrapper *f = NewWrapper();
+
     Printf(f->def, "/* This function is designed to be called by the zend list destructors */\n");
     Printf(f->def, "/* to typecast and do the actual destruction */\n");
     Printf(f->def, "static void %s(zend_resource *res, const char *type_name) {\n", destructorname);
 
-    Printf(f->code, "if(zend_lookup_class(zend_string_init(\"%s\",sizeof(\"%s\")-1,0))) {\n", name, name);
-    Printf(f->code, "return;\n}\n\n");
+    Printf(f->def, "\n\nif(zend_lookup_class(zend_string_init(\"%s\",sizeof(\"%s\")-1,0))) {\n", intermediate_name, intermediate_name);
+    Printf(f->def, "return;\n}\n");
```

Comment:
What's this trying to check?

Looking at the generated code, it seems to be checking "is this type a wrapped class", which we should be able to statically determine that at wrapper generation time, rather than needing a run-time check.

It looks to me like if the type isn't a wrapped class, then the generated code which follows isn't going to compile anyway, but maybe I'm missing something...